### PR TITLE
Support an object for TLS check domains

### PR DESCRIPTION
**Pull request by DinosaurDad**
TLS domains can now either be a string:
  "foo.com"
or an object with options:
  {"domain": "foo.com", "options": {"path": "/robots.txt"}}

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Allow a path (or other options) to be passed to TLS checks, because some sites don't allow the root path to be accessed. OpenSSL can verify the domain, but using the `https` class in NodeJS you must provide a path that is valid.

## How did you implement / how did you fix it  
1.  Added a check for an object or a string, and pass an `options` argument to ssl-check

## How to test  
1. Create certificate domains such as this:
```
  "domains": [
      "foo.com",
      {
        "domain": "bar.com",
        "options": {
          "path": "/robots.txt"
        }
      }
  ]
```

### CODE REVIEW

**Comment by DinosaurDad** [comment]
I cleaned up the code that deals with domains being string or object. This feels a bit more TypeScript (aka, what was I thinking?)

**Review by gsukanto — APPROVED** [review]
lgtm

File: src/components/tls-checker/index.ts [hunk 1]
```diff
@@ -24,15 +24,21 @@
 
 import sslChecker from 'ssl-checker'
 
-export async function checkTLS(url: string, expiryThreshold = 30) {
-  const { valid, validTo, daysRemaining } = await sslChecker(url)
+export async function checkTLS(host: any, expiryThreshold = 30) {
+  const hostOptions = host?.options! ?? {}
+  const domain = host?.domain! ?? host
+
```
Comment (nicnocquee) [inline src/components/tls-checker/index.ts:30]: I think it's easier to understand if we use the exact type for the `host` argument which is either a `string`, or an object with `domain` and `options` properties:

```typescript
type TLSHostArg = {
  domain: string
  options?: Record<string, any>
}
export async function checkTLS(
  host: string | TLSHostArg,
  expiryThreshold = 30
) {
  const hostOptions = (host as TLSHostArg)?.options ?? {}
  const domain = (host as TLSHostArg)?.domain ?? (host as string)
```

File: src/components/tls-checker/index.ts [hunk 2]
```diff
@@ -24,15 +24,21 @@
 
 import sslChecker from 'ssl-checker'
```
Comment (nicnocquee) [inline src/components/tls-checker/index.ts:25]: - [ ] please also add a test in `src/components/tls-checker/index.test.ts`
- [ ] please update the documentation in `docs/src/pages/guides/tls-checkers.md`

**Review by nicnocquee — CHANGES_REQUESTED** [review]

File: src/components/tls-checker/index.ts (hunk 1 above)
Comment (DinosaurDad) [reply src/components/tls-checker/index.ts:30]: I'm not seasoned with TypeScript, so I'm not familiar with the ramifications of that. If that would work here, go for it.

Comment (DinosaurDad) [reply src/components/tls-checker/index.ts]: Done

File: src/components/tls-checker/index.ts (hunk 2 above)
Comment (DinosaurDad) [reply src/components/tls-checker/index.ts:25]: Added and updated.

**Review by nicnocquee — APPROVED** [review]

**Pull request merged**